lib/portage/manifest.py

```diff
@@ -269,40 +264,48 @@ def _parseDigests(self, mylines, myhashdict=None, mytype=None):
 
     def _getDigestData(self, distlist):
         """create a hash dict for a specific list of files"""
-        myhashdict = {}
-        for myname in distlist:
-            for mytype in self.fhashdict:
-                if myname in self.fhashdict[mytype]:
-                    myhashdict.setdefault(mytype, {})
-                    myhashdict[mytype].setdefault(myname, {})
-                    myhashdict[mytype][myname].update(self.fhashdict[mytype][myname])
+        myhashdict = {
+            mytype: {myname: self.fhashdict[mytype][myname]}
+            for myname in distlist
+            for mytype in self.fhashdict
+            if myname in self.fhashdict[mytype]
+        }
         return myhashdict
 
     def _createManifestEntries(self):
         valid_hashes = set(get_valid_checksum_keys())
         valid_hashes.add("size")
-        mytypes = list(self.fhashdict)
-        mytypes.sort()
-        for t in mytypes:
-            myfiles = list(self.fhashdict[t])
-            myfiles.sort()
-            for f in myfiles:
+        mytypes = sorted(self.fhashdict)
+        for mytype in mytypes:
+            myfiles = sorted(self.fhashdict[mytype])
+            for myfile in myfiles:
                 myentry = Manifest2Entry(
-                    type=t, name=f, hashes=self.fhashdict[t][f].copy()
+                    type=mytype,
+                    name=myfile,
+                    hashes=self.fhashdict[mytype][myfile].copy(),
                 )
-                for h in list(myentry.hashes):
-                    if h not in valid_hashes:
-                        del myentry.hashes[h]
+                for myhash in myentry.hashes:
```

Comment:
Could we do an `intersection()` here?

Reply:
After looking into your suggestion and comparing to what was there, what we want is a `difference()`.
```python
>>> foo = [1, 2, 3, 4, 5, 6, 7]
>>> bar = [5, 6, 7, 8, 9, 10]
>>> for f in foo:
...     if f not in bar:
...         print(f)
... 
1
2
3
4
>>> set(foo).difference(bar)
{1, 2, 3, 4}
```

Reply:
You were right, it was an `intersection()`.

However, I'll be pushing up a small goodie bag of changes soon! 🌠

Reply:
Nice!